clang/test/Frontend/sycl-unnamed-lambdas-default.cpp

```diff
@@ -1,18 +1,14 @@
 // RUN: %clang_cc1 -fsycl-is-device %s -verify -DUNNAMED_LAMBDAS
 // RUN: %clang_cc1 -fsycl-is-host %s -verify -DUNNAMED_LAMBDAS
 //
-// RUN: %clang_cc1 -fsycl-is-host -sycl-std=2017 %s -verify
+// RUN: %clang_cc1 -fsycl-is-host -sycl-std=2020 %s -verify
 // RUN: %clang_cc1 -fsycl-is-host -sycl-std=2020 %s -verify -DUNNAMED_LAMBDAS
 //
 // RUN: %clang_cc1 -fsycl-is-host -fno-sycl-unnamed-lambda %s -verify
-// RUN: %clang_cc1 -fsycl-is-host -sycl-std=2017 -fno-sycl-unnamed-lambda %s -verify
-// RUN: %clang_cc1 -fsycl-is-host -sycl-std=2017 -fsycl-unnamed-lambda %s -verify -DUNNAMED_LAMBDAS
+// RUN: %clang_cc1 -fsycl-is-host -sycl-std=2020 -fno-sycl-unnamed-lambda %s -verify
+// RUN: %clang_cc1 -fsycl-is-host -sycl-std=2020 -fsycl-unnamed-lambda %s -verify -DUNNAMED_LAMBDAS
 
 // expected-no-diagnostics
 #if defined(UNNAMED_LAMBDAS) && !defined(__SYCL_UNNAMED_LAMBDA__)
-#error "Unnamed lambdas should be enabled for SYCL2020/default"
-#endif
-
-#if !defined(UNNAMED_LAMBDAS) && defined(__SYCL_UNNAMED_LAMBDA__)
-#error "Unnamed lambdas should NOT be enabled for SYCL2017, or when explicitly disabled"
```

Comment:
This `#if` should be preserved, but we may want to tweak the message here. For `RUN` lines 7 and 8 we want unnamed lambas to be disabled and without this `#if` there will be no check if that actually happens.